File: sample/src/main/java/ru/slybeaver/truecalendar/MainActivity.java
package ru.slybeaver.truecalendar;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Locale;

import ru.slybeaver.slycalendarview.SlyCalendarDialog;

public class MainActivity extends AppCompatActivity implements SlyCalendarDialog.Callback {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.btnShowCalendar).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new SlyCalendarDialog()
                        .setSingle(false)
                        .setFirstMonday(false)
                        .setCallback(MainActivity.this)
                        .show(getSupportFragmentManager(), "TAG_SLYCALENDAR");
            }
        });
    }

    @Override
    public void onCancelled() {
        //Nothing
    }

    @Override
    public void onDataSelected(Calendar firstDate, Calendar secondDate, int hours, int minutes) {
        if (firstDate != null) {
            if (secondDate == null) {
                firstDate.set(Calendar.HOUR_OF_DAY, hours);
                firstDate.set(Calendar.MINUTE, minutes);
                Toast.makeText(
                        this,
                        new SimpleDateFormat(getString(R.string.timeFormat), Locale.getDefault()).format(firstDate.getTime()),
                        Toast.LENGTH_LONG

                ).show();
            } else {
                Toast.makeText(
                        this,
                        getString(
                                R.string.period,
                                new SimpleDateFormat(getString(R.string.dateFormat), Locale.getDefault()).format(firstDate.getTime()),
                                new SimpleDateFormat(getString(R.string.timeFormat), Locale.getDefault()).format(secondDate.getTime())
                        ),
                        Toast.LENGTH_LONG

                ).show();
            }
        }
    }
}

File: sample/src/main/java/ru/slybeaver/truecalendar/MainActivity.kt
package ru.slybeaver.truecalendar

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.snackbar.Snackbar
import ru.slybeaver.slycalendarview.SlyCalendarDialog
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by danielvilha on 28/05/20
 */
class MainActivity : AppCompatActivity(), SlyCalendarDialog.Callback {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<View>(R.id.btnShowCalendar).setOnClickListener {
            SlyCalendarDialog()
                    .setSingle(false)
                    ?.setFirstMonday(false)
                    ?.setCallback(this@MainActivity)
                    ?.show(supportFragmentManager, "TAG_SLYCALENDAR")
        }

        findViewById<View>(R.id.btnShowCalendarHidden).setOnClickListener {
            SlyCalendarDialog()
                .setSingle(false)
                ?.setFirstMonday(false)
                ?.showTime(false)
                ?.setCallback(this@MainActivity)
                ?.show(supportFragmentManager, "TAG_SLYCALENDAR")
        }
    }

    override fun onCancelled() {
        Toast.makeText(this, "Cancelled", Toast.LENGTH_LONG).show()
    }

    override fun onDataSelected(firstDate: Calendar?, secondDate: Calendar?, hours: Int, minutes: Int) {
        if (firstDate != null) {
            if (secondDate == null) {
                firstDate[Calendar.HOUR_OF_DAY] = hours
                firstDate[Calendar.MINUTE] = minutes
                Snackbar.make(
                    findViewById(R.id.btnShowCalendar),
                    SimpleDateFormat(getString(R.string.timeFormat), Locale.getDefault()).format(firstDate.time),
                    Snackbar.LENGTH_LONG
                ).show()
            } else {
                secondDate[Calendar.HOUR_OF_DAY] = hours
                secondDate[Calendar.MINUTE] = minutes
                Snackbar.make(
                    findViewById(R.id.btnShowCalendar),
                    getString(
                        R.string.period,
                        SimpleDateFormat(getString(R.string.dateFormat), Locale.getDefault()).format(firstDate.time),
                        SimpleDateFormat(getString(R.string.timeFormat), Locale.getDefault()).format(secondDate.time)
                    ),
                    Snackbar.LENGTH_LONG
                ).show()
            }
        } else {
            Snackbar.make(
                findViewById(R.id.btnShowCalendar),
                "No days selected",
                Snackbar.LENGTH_LONG
            ).show()
        }
    }
}